alf/algorithms/ddpg_algorithm.py

```diff
@@ -256,25 +259,29 @@ def _critic_train_step(self, exp: Experience, state: DdpgCriticState):
         return state, info
 
     def _actor_train_step(self, exp: Experience, state: DdpgActorState):
+        info = {}
         action, actor_state = self._actor_network(
-            exp.observation, state=state.actor)
+            exp.observation, state=state.actor, info=info)
 
         q_values, critic_states = self._critic_networks(
             (exp.observation, action), state=state.critics)
         q_value = q_values.min(dim=1)[0]
 
         dqda = nest_utils.grad(action, q_value.sum())
 
-        def actor_loss_fn(dqda, action):
+        def actor_loss_fn(dqda, action, pre_activation):
             if self._dqda_clipping:
                 dqda = torch.clamp(dqda, -self._dqda_clipping,
                                    self._dqda_clipping)
             loss = 0.5 * losses.element_wise_squared_loss(
                 (dqda + action).detach(), action)
             loss = loss.sum(list(range(1, loss.ndim)))
+            if self.action_l2 > 0:
```

Comment:
Prefix `action_l2` with "_". And just need to compute a cost based on the current `action`